src/value-types.js

```diff
@@ -0,0 +1,218 @@
+var execute = require('./execute');
+
+var Value = function Value(value) {
+    this.value = value;
+};
+
+Value.prototype.equals = function(b) {
```

Comment:
That's true, it stops you using `instanceof` (well, you can set `constructor`, but there's no hierarchy).